_minutes/2024-11-07-wecg.md

```diff
@@ -0,0 +1,184 @@
+ * [timothy] We have closed tab groups as well.
+ * [simeon] Would it make sense to have closed groups be a distinct type of tab to query?
+ * [timothy] Not necessarily.
+ * [rob] How does this functionality relate to sessions.restore? Based on the description it sounds similar.
+ * [oliver] Pinned fields sounds non-controversial, querying/managing closed groups sounds like something needing more discussion.
+
+[Issue 716](https://github.com/w3c/webextensions/issues/716): Inconsistent support for [context]menus.getTargetElement()
+
+ * [rob] Feature request for the ability to retrieve the element that is clicked when a context menu was opened. Firefox solved this by introducing getTargetElement(). When contextMenus.onClicked fires you can get a reference (targetElementId) that you pass to the menus.getTargetElement function in a content script to retrieve the element. There are workarounds such as exposing the coordinates in the document, but they're lossy. Firefox's approach is a reliable solution.
+ * [timothy] Against coordinate suggestion as well; interested in implementing Firefox's getTargetElement API.
+ * [oliver] Treating this as opaque is fine, but I'm curious what this ID looks like in Firefox.
+ * [rob] It's generated in Firefox. I believe it's the timestamp. On click, a weak ref to the element is stored, timestamp is captured and passed to the extension.
+ * [oliver] Thanks, just wanted to understand the general shape.
+ * [timothy] I'd want it to be opaque. UUID or some string.
+ * [rob] It's currently an integer. This enables extensions to pass the value to getTargetElement in a code string (tabs.executeScript) without serialization concerns. That concern is lesser in MV3 because scripting.executeScript does not accept code strings, and accepts args for func instead.
+ * [timothy] Want it to be opaque so there's no meaning associated with it. Allows us to modify the implementation later if necessary. E.g. across processes, etc.
+ * [rob] I don't think that it should be a string; the integer can be a random number, as it is only used to avoid race conditions where “the last clicked element” changed between the time of the invocation of the context menu and the invocation of the script.
+ * [tomislav] I don't think Timothy means a value you can't look into, just a value without meaning.
+ * [timothy] Exactly. Not a timestamp. No inherent meaning to the value.
+ * [rob] Thanks for the clarification - I thought that you meant a string or object by “opaque”. An integer sounds good to me.
+ * [rob] Safari seems on board. Does Chrome have any oppositions?
+ * [oliver] I can follow up.
+
+[Issue 717](https://github.com/w3c/webextensions/issues/717): Request: [ContextMenus] Include `<canvas>`s in the "page" ContextType (or it's own ContextType)
+
+ * [simeon] The only way to match &lt;canvas> elements is to specify “all”. The request is to offer a specific contextType to match &lt;canvas>.
```

Comment:
```suggestion
 * [simeon] The only way to match `<canvas>` elements is to specify “all”. The request is to offer a specific `contextType` to match `<canvas>`.
```

Reply:
`&lt;` is a valid way to render `<` in markdown, at least on Github.

I have applied this suggestion since the title is also using backticks around the canvas; but in the future let's not wrap `<canvas>` unnecessarily. I try to minimize the amount of unnecessary formatting in the notes because it may be distracting from its content.